### ISSUE TITLE
New Rule: `no-redundant-else`

### ISSUE THREAD

**Issue by Jay-Karia** [eslint/eslint#18949]
### Rule details

Enforces to remove extra else statement when not needed.

### Related ECMAScript feature

N/A

### What type of rule is this?

Suggests an alternate way of doing something

### Example code
``` js
// invalid
function func_name(valid) {
    if (valid) {
        return true
    } else {
        return false
    }
}

// valid
function func_name(valid) {
    if (valid) {
        return true
    }

    return false
}
```

### Why should this rule be in the core instead of a plugin?

This rule relates to core JS coding style and do not think this will fit well in plugins.

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

I will take some time to implement this (if accepted), as I am new to this.

**Comment by snitin315** [comment]
We already [`consistent-return`](https://eslint.org/docs/latest/rules/consistent-return#rule-details) rule in place, and I think this should suffice your use case.

**Comment by aladdin-add** [comment]
it's the rule https://eslint.org/docs/latest/rules/no-else-return you are looking for. :)

**Comment by Jay-Karia** [comment]
Oh, my bad
